pulsar-metadata/src/test/java/org/apache/pulsar/metadata/bookkeeper/EndToEndTest.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.metadata.bookkeeper;
+
+import java.nio.charset.StandardCharsets;
+import java.util.function.Supplier;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.bookkeeper.client.BookKeeper;
+import org.apache.bookkeeper.client.api.DigestType;
+import org.apache.bookkeeper.client.api.LedgerEntries;
+import org.apache.bookkeeper.client.api.LedgerEntry;
+import org.apache.bookkeeper.client.api.ReadHandle;
+import org.apache.bookkeeper.client.api.WriteHandle;
+import org.apache.pulsar.metadata.BaseMetadataStoreTest;
+import org.testng.annotations.Test;
+
+/**
+ * Test the zookeeper implementation of the ledger replication manager.
+ */
+@Slf4j
+public class EndToEndTest extends BaseMetadataStoreTest {
+    @Test(dataProvider = "impl")
+    public void testBasic(String provider, Supplier<String> urlSupplier) throws Exception {
+        @Cleanup
+        BKTestCluster bktc = new BKTestCluster(urlSupplier.get(), 1);
+
+        @Cleanup
+        BookKeeper bkc = bktc.newClient();
+
+        long ledgerId;
+        {
+            @Cleanup
+            WriteHandle wh = bkc.newCreateLedgerOp()
```

Comment:
What about adding a minimal test about opening the ledger in recovery mode?

Reply:
Good point. Added one.